release-content/0.16/release-notes/17398_Relationships_nonfragmenting_onetomany.md

```diff
@@ -1,4 +1,63 @@
 
-<!-- TODO -->
+This worked, but it had some pretty glaring downsides:
+
+1. Maintaining hierarchies was "separate" from the core ECS data model. This made it hard to improve our spawn APIs, and made interacting with hierarchies less natural.
+2. The system was specialized and not reusable. Developers wanting to define their own relationship types had to reinvent the wheel.
+3. To ensure data integrity, expensive scans were required to avoid duplicates.
+
+In **Bevy 0.16** we have added initial support for Relationships: a generalized and efficient component-driven system for linking entities together bidirectionally. This is what defining a new [`Relationship`] looks like:
+
+```rust
+/// This is a "relationship" component.
+/// Add it to an entity that "likes" another entity.
+#[derive(Component)]
+#[relationship(relationship_target = LikedBy)]
+struct Likes(pub Entity);
+
+/// This is the "relationship target" component.
+/// It will be automatically inserted and updated to contain
+/// all entities that currently "like" this entity.
+#[derive(Component, Deref)]
+#[relationship_target(relationship = Likes)]
+struct LikedBy(Vec<Entity>);
+
+// Later in your app
+let e1 = world.spawn_empty().id();
+let e2 = world.spawn(Likes(e1)).id();
+let e3 = world.spawn(Likes(e1)).id();
+
+// e1 is liked by e2 and e3 
+let liked_by = world.entity(e1).get::<LikedBy>().unwrap();
+assert_eq!(&**liked_by, &[e2, e3]);
+```
+
+The [`Relationship`] component is the "source of truth", and the [`RelationshipTarget`] component is updated to reflect that source of truth. This means that adding/removing relationships should always be done via the [`Relationship`] component.
+
+We use this "source of truth" model instead of allowing both components to "drive" for performance reasons. Allowing writes to both sides would require expensive scanning during inserts ensure they are in sync and have no duplicates. The "relationships as the source of truth" approach allows us to make adding relationships constant-time (which is an improvement over previous Bevy versions!).
+
+Relationships are built on top of Bevy's [Component Hooks](/news/bevy-0-14/#ecs-hooks-and-observers), which immediately and efficiently maintains the connection between the [`Relationship`] and the [`RelationshipTarget`] by plugging directly into the component add/remove/update lifecycle. In combination with the new Immutable Components feature (relationship components are immutable), this ensures data integrity is maintained no matter what developers do!
```

Comment:
Might want a link to the Immutable Components section too (not sure if it's written yet)